lib/matplotlib/scale.py

```diff
@@ -82,6 +82,84 @@ def get_transform(self):
         return IdentityTransform()
 
 
+class ArbitraryTransform(Transform):
+     """
+     A simple transform that takes and arbitrary function for the
+     forward and inverse transform.
+     """
+
+     input_dims = 1
+     output_dims = 1
+     is_separable = True
+     has_inverse = True
+
+     def __init__(self, forward, inverse):
+         """
+         Parameters
+         ----------
+
+         forward: The forward function for the transform
```

Comment:
Should be something like

~~~
forward : callable
    The forward function for the transform. It must have the signature::

        def forward(values: array-like) -> array-like
~~~

Are there any additional constraints? E.g. must the function be monotonic? Do we need more precise type descriptions?

Reply:
Changed to 
```
         forward: callable
             The forward function for the transform.  This function must have
             an inverse and, for best behavior, be monotonic. 
             It must have the signature::

                def forward(values: array-like) ->
                        array-likeThe forward function for the transform

         inverse: callable
            The inverse of the forward function.  Signature as ``forward``.
```